ocp_resources/machineset.py

```diff
@@ -0,0 +1,225 @@
+import logging
+
+from ocp_resources.resource import NamespacedResource, ResourceEditor
+from ocp_resources.utils import TimeoutExpiredError, TimeoutSampler
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+TIMEOUT = 300
+
+
+class MachineSetNotReadyError(Exception):
+    pass
+
+
+class MachineSet(NamespacedResource):
+    """
+    Machineset object.
+
+    Args:
+        cluster_name (str): OpenShift cluster name.
+        machine_role (str): machine role. e.g.: 'worker'.
+        machine_type (str): machine role. e.g.: 'worker'.
+        replicas (int): amount of replicas the machine-set will have.
+        provider_spec (dict): provider spec information.
+        wait_ready_state (bool): True if to wait for machine-set to be 'ready' state, False otherwise.
+
+        provider spec example:
+
+    {
+        "value": {
+            "apiVersion": "ovirtproviderconfig.machine.openshift.io/v1beta1",
+            "auto_pinning_policy": "none",
+            "cluster_id": 5612af70-f4f5-455d-b7df-fbad66accc38,
+            "cpu": {
+                "cores": 8,
+                "sockets": 1,
+                "threads": 1
+            },
+            "credentialsSecret": {
+                "name": "ovirt-credentials"
+            },
+            "kind": "OvirtMachineProviderSpec",
+            "memory_mb": 16000,
+            "os_disk": {
+                "size_gb": 31
+            },
+            "template_name": "ge2n1-gcwmg-rhcos",
+            "type": "server",
+            "userDataSecret": {
+                "name": "worker-user-data"
+            }
+        }
+    }
+    """
+
+    api_group = NamespacedResource.ApiGroup.MACHINE_OPENSHIFT_IO
+
+    def __init__(
+        self,
+        cluster_name=None,
+        name=None,
+        namespace=None,
+        teardown=True,
+        client=None,
+        machine_role="worker",
+        machine_type="worker",
+        replicas=1,
+        provider_spec=None,
+        yaml_file=None,
+        wait_ready_state=True,
+    ):
+        super().__init__(
+            name=name,
+            namespace=namespace,
+            client=client,
+            teardown=teardown,
+            yaml_file=yaml_file,
+        )
+        self.replicas = replicas
+        self.cluster_name = cluster_name
+        self.machine_role = machine_role
+        self.machine_type = machine_type
+        self.provider_spec = provider_spec or {}
+        self.wait_ready_state = wait_ready_state
+
+    def __enter__(self):
+        super().__enter__()
+        if self.wait_ready_state and not self.wait(timeout=self.replicas * TIMEOUT):
```

Comment:
This is not part of the creation, if you need to wait for some state, do it in a function.
__enter__ is called when the user do `with ....`

Reply:
ack